test/integration/targets/filter_core/tasks/password_hash.yml

```
@@ -87,3 +87,58 @@
       assert:
         that:
           - yescrypt_hash_salt == '$y$j9T$V7qMYJaNbVKOeh4PhtqPk/$x7rTqZ.RpI07.dkBSxcg.jLM8ODUfx25rCN0cFsAUg0'
+
+    - name: Test salt with invalid character
+      set_fact:
+        md5_hash: "{{ 'test' | password_hash('md5_crypt', salt='123^') }}"
+      ignore_errors: yes
+      register: bad_salt_result
+
+    - name: Determine if bad salt character was recognized
+      assert:
+        that:
+          - "'invalid characters in salt' in bad_salt_result.msg"
```

I wonder if we could just combine simple one-step tests with assertions
```suggestion
      register: bad_salt_result
      failed_when: >-
        bad_salt_result is not failed or
        'invalid characters in salt' not in bad_salt_result.msg
```

Should we have these coupled? WDYT? I've found that when steps are spread out, it may be difficult to trace which of multiple tasks being a part of one logical test.

---

This style does not seem to work in this instance, though honestly, I'd expect it to.

---

Yeah, in-browser coding doesn't reveal all nuance..